File: Java/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    String[] jokaBabb = new String[]{"aya", "ye", "woo", "ma"};
    
    public boolean checkCanSpeak(String word) {
        for (String j: jokaBabb){
            if (word.length() == 0) return true;
            if (word.startsWith(j)){
                return checkCanSpeak(word.substring(j.length()));
            }
        }
        return false;
    }
    
    public int solution(String[] babbling) {
        int answer = 0;
        for (String b: babbling) {
            if (checkCanSpeak(b)) {
                answer++;
            }
        }
        return answer;
    }
}

File: Python3/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.py
jokaBabb = ["aya", "ye", "woo", "ma"]
def checkCanSpeak(word):
    for j in jokaBabb:
        if len(word) == 0:
            return True
        if word.startswith(j):
            return checkCanSpeak(word[len(j):])
    return False

def solution(babbling):
    answer = 0
    for b in babbling:
        if checkCanSpeak(b):
            answer += 1
    return answer